### PR TITLE
Improve update-locales script and fix locale processing bug

### PR DESCRIPTION
**Pull request by wxiaoguang**
The locales of Gitea has been broken for long time, till now, it's still not fully fixed.

One of the root problems is that the `ini` library is quite quirky and the `update-locales` script doesn't work well for all cases.

This PR fixes the  `update-locales` script to make it satisfy `ini` library and the crowdin.

See the comments for more details.

The `locale_zh-CN.ini` is an example, it comes from crowdin and is processed by the new `update-locales.sh`. Especially see the `feed_of`: https://github.com/go-gitea/gitea/pull/23240/files#diff-321f6ca4eae1096eba230e93c4740f9903708afe8d79cf2e57f4299786c4528bR268

### CODE REVIEW

**Review by wolfogre — APPROVED** [review]

File: build/update-locales.sh
Comment (wxiaoguang) [inline build/update-locales.sh]:
```suggestion
# crowdin always outputs quoted strings if there is quotes in the strings.
```

**Review by zeripath — APPROVED** [review]
Thanks @wxiaoguang 

I kept meaning to get round to looking at this as I suspected that something was going wrong to do with the ini output preparation.

This to me indicates again that we either need to get on with fixing go-ini or get off it.

**Comment by zeripath** [comment]
I'd also say we might want to consider making this script some go code instead

**Review by delvh — APPROVED** [review]

**Comment by wxiaoguang** [comment]
> I'd also say we might want to consider making this script some go code instead

Agree, and I prefer to get rid of the `ini` more 😁 if there is no buggy ini package, we do not need these tricks.

I think this PR is the last fix by modifying the script. Next time we should decide whether use Go to rewrite or totally replace the `ini` package.

File: build/update-locales.sh
Comment (lunny) [inline build/update-locales.sh]: remove ?

Comment (wxiaoguang) [reply build/update-locales.sh]:
```suggestion
# * remove the leading quote
```

Yup, typo ....

**Comment by lunny** [comment]
replace #23239

**Comment by jolheiser** [comment]
🎺 🤖

File: build/update-locales.sh
Comment (silverwind) [inline build/update-locales.sh:10]: Could just do the same detection that `Makefile` does and adapt `sed` command syntax based on whether it is GNU or BSD sed. Better than to require the installation of additional software :)

Comment (silverwind) [reply build/update-locales.sh:10]: I'll check it out.

Comment (silverwind) [reply build/update-locales.sh:10]: https://github.com/go-gitea/gitea/pull/23254

Comment (wxiaoguang) [reply build/update-locales.sh:10]: Thank you. Actually I have thought about this problem, but I didn't choose the `sed -i''` for BSD sed, because the script has been here for long time, nobody really needs it, I just want to avoid any inconsistency. And I think this PR is the last fix by modifying the script 😂 in the future  we should decide whether use Go to rewrite or totally replace the ini package.

The `sed -i''` might also work, just like it in Makefile, but  I haven't tested it.

**Comment by wxiaoguang** [comment]
I didn't know where this script runs in. If it runs in the environment without bash (really? it's 2023 now ...), I think we should also revert the shebang line to #!/bin/sh and remove the bash related `if [[`.

ps: I didn't see today's crowdin sync commit, does it mean that this script failed because I used bash in last PR?
I want to mention some maintainers but I don't know who knows the crwodin sync infrastructure ....

File: build/update-locales.sh
Comment (wxiaoguang) [reply build/update-locales.sh:10]: Some more explanations about my "worry about inconsistency": this script is hard to debug, and not covered by any  test (unlike the sed in Makefile, any error can be found in the first time and gets fixed quickly). If there is any inconsistency between GNU sed and BSD sed, the problem can not be found until next crowdin sync runs, and it's difficult for macOS users to debug the inconsistency if  they are using a different version.

If there is no inconsistency, then using `sed -i''` is fine for BSD sed.

Just share some of my thoughts, for your information, no objection nor block. 😁

**Comment by wxiaoguang** [comment]
Sorry, I think it's incorrect to use `bash` in this script. 

The last run fails:

https://drone.gitea.io/go-gitea/gitea/68658/1/3

```
3.17: Pulling from library/alpine
Digest: sha256:69665d02cb32192e52e07644d76bc6f25abeb5410edc1c7a81a10ba3f0efb90a
Status: Image is up to date for alpine:3.17
+ ./build/update-locales.sh
/bin/sh: ./build/update-locales.sh: not found
```

**Pull request merged**
